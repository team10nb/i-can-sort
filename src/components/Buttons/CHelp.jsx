import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import IconButton from '@material-ui/core/IconButton';
import Avatar from '@material-ui/core/Avatar';
import Modal from '@material-ui/core/Modal';
import Backdrop from '@material-ui/core/Backdrop';
import Fade from '@material-ui/core/Fade';
import Tooltip from '@material-ui/core/Tooltip';
import HelpOutlineOutlinedIcon from '@material-ui/icons/HelpOutlineOutlined';
import Button from '@material-ui/core/Button';
import CardHeader from '@material-ui/core/CardHeader';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import Typography from '@material-ui/core/Typography';
import HighlightOffIcon from '@material-ui/icons/HighlightOff';
import Dialog from '@material-ui/core/Dialog';
import logo1 from '../../Resource/Chelp1.png';
import logo2 from '../../Resource/Chelp2.png';
import logo3 from '../../Resource/Chelp3.png';
import {color} from '../../scenes/mainPages/Correctness';

{/*
  This function is for designing the set and help button  
*/}

  const useStyles = makeStyles((theme) => ({
    modal: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'center',
    },
    card: {
      maxWidth: 580,
      minHeight: 560,
    },
    avatar: {
      backgroundColor: color,
    },
    button: {
      marginRight: theme.spacing(1),
      
    },
    button1:
    {
      marginRight: theme.spacing(1),
      backgroundColor:color,
    },
  }));
  
  

  

  function getStepContent(step) {
    switch (step) {
      case 0:
        return (
          <CardActionArea >
            <img src={logo1} alt="logo1" width='587' heigh='300'  />
            <CardContent>
              <Typography variant="body2" color="textSecondary" component="p">
                This a entry for each sorting algorithm.
                Choose one and click it to enter next Page.
                A progress bar is at the bottom, it shows how much you have learnt.      
              </Typography>                
            </CardContent>
          </CardActionArea>

        );
      case 1:
        return (
          <CardActionArea >
            <img src={logo2} alt="logo2" width='587' heigh='300'  />
            <CardContent>
              <Typography variant="body2" color="textSecondary" component="p">
                Left one is a setting button, click  it to change settings.
                Right one is a help button, click it to see explanations for each component.          
              </Typography>             
            </CardContent>
          </CardActionArea>
        );
      default:
        return 'Unknown step';
    }
  }


 
  export  default function CHelp() {
    const classes = useStyles();

    const [open, setOpen] = React.useState(false);
 

    const handleClickOpen = () => {
      setOpen(true);
    };

    const handleClose = () => {
      setOpen(false);
    };


    const [activeStep, setActiveStep] = React.useState(0);  

    const steps = 3;
    
    const handleNext = () => {
    setActiveStep((prevActiveStep) => prevActiveStep + 1);     
    };

    const handleBack = () => {
      setActiveStep((prevActiveStep) => prevActiveStep - 1);
    };
    const handleReset = () => {
      setActiveStep(0);
    };

    const card =(
      <Dialog onClose={handleClose} aria-labelledby="simple-dialog-title" open={open}>
        <Fade in={open}>
          <Card className={classes.card}>
            <CardHeader
              avatar={
                <IconButton aria-label="close"  onClick={handleClose}>
                <HighlightOffIcon  style={{ color: color, fontSize: 30  }} />
              </IconButton>
              }
            />
              {activeStep === steps - 1 ? (
              <div>
                <CardActionArea  >               
                  <img src={logo3} alt="logo3" width='587' heigh='300'  />
                  <CardContent>
                    <Typography variant="body2" color="textSecondary" component="p">
                      This a choice menu. Menu contains Tutorial, Procedure, Efficiency, Correctness and Exercise.
                      Choose a module and click it.   
                    </Typography>                
                  </CardContent>
                  
                </CardActionArea>
                <CardActions>
                  <Button onClick={handleReset} variant="outlined" style={{ color: color}}>Reset</Button>
                </CardActions>
              </div>
              ) : (
               
                  <div>
                    {getStepContent(activeStep)}                    
                      <CardActions>
                        <Button disabled={activeStep === 0} onClick={handleBack} variant="outlined" style={{ color: color}}>Back</Button>                     
                        <Button
                          variant="contained"
                          color= {color}
                          onClick={handleNext}
                          className={classes.button1}
                        >
                          Next Tip
                        </Button>
                      </CardActions>                 
                  </div>
                )}
          </Card> 
        </Fade>
      </Dialog>   
    );

    return(
      <div>
        <Tooltip title="tutorial" placement="bottom" arrow>
          <IconButton aria-label="help"  onClick={handleClickOpen}>
            <HelpOutlineOutlinedIcon />
          </IconButton>
        </Tooltip>
        {card}     
      </div>
    )
  }

