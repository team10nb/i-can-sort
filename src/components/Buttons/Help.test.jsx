import { render } from '@testing-library/react';
import CHelp from './CHelp';

const mockHandleClick = jest.fn();
    

const props = {
    color: '#1565c0',
    handleOpen: mockHandleClick,
};
test('should contain 1 buttons', () => {
    const {getAllByRole} = render(<CHelp {...props}/>);
    const buttons = getAllByRole('button');
    expect(buttons.length).toBe(1);
})

test('help button should match Snapshot', () => {
    
 
const content = render(<CHelp {...props} />);
expect(content).toMatchSnapshot();
})