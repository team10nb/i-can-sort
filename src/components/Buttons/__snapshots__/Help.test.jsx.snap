// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`help button should match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          style="height: 48px;"
        >
          <button
            aria-label="help"
            class="MuiButtonBase-root MuiIconButton-root"
            tabindex="0"
            title="help"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root makeStyles-icon-9"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <div
            aria-hidden="true"
            class="MuiBackdrop-root makeStyles-backdrop-7"
            style="opacity: 0; visibility: hidden;"
          >
            <div
              class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                style="margin-left: -490px; margin-top: 10px;"
              >
                <button
                  aria-label="close"
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="color: rgb(21, 101, 192); opacity: 0.8; font-size: 40px;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div>
                <div>
                  <button
                    class="MuiButtonBase-root MuiCardActionArea-root"
                    style="height: 415px; margin-bottom: 15px;"
                    tabindex="0"
                    type="button"
                  >
                    <img
                      alt="logo0"
                      width="587px"
                    />
                    <div
                      class="MuiCardContent-root"
                    >
                      <p
                        class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                        style="margin-left: 20px; margin-right: 20px;"
                      >
                        This a entry for each sorting algorithm. Choose one and click it to enter next Page. A progress bar is at the bottom, it shows how much you have learnt.
                      </p>
                    </div>
                    <span
                      class="MuiCardActionArea-focusHighlight"
                    />
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <div
                  class="makeStyles-button-11"
                >
                  <button
                    class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                    disabled=""
                    style="background-color: rgb(21, 101, 192); opacity: 0.5;"
                    tabindex="-1"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        style="color: white; font-size: 40px;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                        />
                      </svg>
                    </span>
                  </button>
                  <button
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                    style="background-color: rgb(21, 101, 192); opacity: 0.5;"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        style="color: white; font-size: 40px;"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                        />
                      </svg>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        style="height: 48px;"
      >
        <button
          aria-label="help"
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          title="help"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-9"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <div
          aria-hidden="true"
          class="MuiBackdrop-root makeStyles-backdrop-7"
          style="opacity: 0; visibility: hidden;"
        >
          <div
            class="MuiPaper-root MuiCard-root makeStyles-card-8 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              style="margin-left: -490px; margin-top: 10px;"
            >
              <button
                aria-label="close"
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    style="color: rgb(21, 101, 192); opacity: 0.8; font-size: 40px;"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <div>
              <div>
                <button
                  class="MuiButtonBase-root MuiCardActionArea-root"
                  style="height: 415px; margin-bottom: 15px;"
                  tabindex="0"
                  type="button"
                >
                  <img
                    alt="logo0"
                    width="587px"
                  />
                  <div
                    class="MuiCardContent-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
                      style="margin-left: 20px; margin-right: 20px;"
                    >
                      This a entry for each sorting algorithm. Choose one and click it to enter next Page. A progress bar is at the bottom, it shows how much you have learnt.
                    </p>
                  </div>
                  <span
                    class="MuiCardActionArea-focusHighlight"
                  />
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="makeStyles-button-11"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root Mui-disabled MuiIconButton-sizeSmall Mui-disabled"
                  disabled=""
                  style="background-color: rgb(21, 101, 192); opacity: 0.5;"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="color: white; font-size: 40px;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                      />
                    </svg>
                  </span>
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeSmall"
                  style="background-color: rgb(21, 101, 192); opacity: 0.5;"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      style="color: white; font-size: 40px;"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                      />
                    </svg>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
